app/utils/TaskListHelper.scala

```diff
@@ -79,25 +82,42 @@ class TaskListHelper(journey: Option[UserAnswers])(implicit messages: Messages)
         controllers.register.routes.DeclarationController.onPageLoad().url))
     else None
 
-  private def declarationEnabled(implicit userAnswers: UserAnswers): Boolean =
-    (userAnswers.get(IsAboutSchemeCompleteId), userAnswers.get(IsWorkingKnowledgeCompleteId)) match {
-      case (Some(true), Some(true)) if userAnswers.allEstablishersAfterDelete.forall(_.isCompleted) &&
-        userAnswers.allTrusteesAfterDelete.forall(_.isCompleted) => true
-      case _ => false
-    }
-
   private def listOf(sections: Seq[Entity[_]]): Seq[JourneyTaskListSection] =
-    for (section <- sections) yield
-      JourneyTaskListSection(
-        Some(section.isCompleted),
-        Link(linkText(section),
-          section.editLink),
-        Some(section.name))
+    for (section <- sections) yield {
+      JourneyTaskListSection( Some(section.isCompleted), Link(linkText(section), section.editLink), Some(section.name))
+    }
 
   private def linkText(item: Entity[_]): String =
     item.id match {
       case EstablisherCompanyDetailsId(_) | TrusteeCompanyDetailsId(_) => messages("messages__schemeTaskList__company_link")
       case EstablisherDetailsId(_) | TrusteeDetailsId(_) => messages("messages__schemeTaskList__individual_link")
       case EstablisherPartnershipDetailsId(_) | TrusteePartnershipDetailsId(_) => messages("messages__schemeTaskList__partnership_link")
     }
+
+  private def flagValue[A](flag : TypedIdentifier[A])(implicit rds: Reads[A]): Option[A] ={
+    journey.flatMap(x => x.get[A](flag))
+  }
+
+  private def isAllEstablishersCompleted(userAnswers: UserAnswers) : Boolean = {
+    userAnswers.allEstablishersAfterDelete.nonEmpty && userAnswers.allEstablishersAfterDelete.forall(_.isCompleted)
+  }
+
+  private def isAllTrusteesCompleted(userAnswers: UserAnswers) : Boolean = {
+
+    val listOfSchemeTypeTrusts: Seq[SchemeType] = Seq(SchemeType.SingleTrust, SchemeType.MasterTrust)
+    val isTrusteesOptional = flagValue(SchemeDetailsId).forall(scheme => !listOfSchemeTypeTrusts.contains(scheme.schemeType))
+
+    val isOptionalTrusteesJourney = (flagValue(HaveAnyTrusteesId).forall(_==false) && isTrusteesOptional)
```

Comment:
No need for outer brackets. Same on following line too.

Reply:
fixed